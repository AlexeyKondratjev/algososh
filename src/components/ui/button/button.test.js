import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';


describe('Тесты компонента Button.', () => {
    it('Кнопка с текстом рендерится без ошибок', () => {
        const button = renderer.create(<Button text="Выполнить" />).toJSON();
        expect(button).toMatchSnapshot();
    });

    it('Кнопка без текста рендерится без ошибок', () => {
        const button = renderer.create(<Button />).toJSON();
        expect(button).toMatchSnapshot();
    });

    it('Заблокированная кнопка рендерится без ошибок', () => {
        const button = renderer.create(<Button disabled />).toJSON();
        expect(button).toMatchSnapshot();
    });

    it('Кнопка с индикацией загрузки рендерится без ошибок', () => {
        const button = renderer.create(<Button isLoader />).toJSON();
        expect(button).toMatchSnapshot();
    });

    it('Корректно отрабатывает колбек при клике на кнопку', () => {
        window.alert = jest.fn();

        render(<Button text='Выполнить' onClick={alert('Кнопка была нажата!')} />);

        const button = screen.getByText("Выполнить");

        fireEvent.click(button);

        expect(window.alert).toHaveBeenCalledWith('Кнопка была нажата!');
    });
});
