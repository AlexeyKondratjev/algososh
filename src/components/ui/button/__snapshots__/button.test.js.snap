// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Button. Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тесты компонента Button. Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тесты компонента Button. Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тесты компонента Button. Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Выполнить
  </p>
</button>
`;
